src/client/opamCommands.ml

```diff
@@ -408,8 +416,16 @@ let init cli =
         OpamStd.Sys.guess_dot_profile shell >>| OpamFilename.of_string)
     in
     let cygwin_setup =
+      let cygwin_internal =
+        match cygwin_internal, cygwin_packages with
+        | `internal, Some pkgs -> `internal_with_pkgs (List.map OpamSysPkg.of_string pkgs)
+        | _, Some _ ->
+          OpamConsole.error_and_exit `Bad_arguments
+            "Option --cygwin-packages is only compatible with --cygwin-internal-install option";
```

Comment:
Note that `--cygwin-location` below is _ignored_ if ``` `none``` is given, but I think this is actually better (with the handling for ``` `none``` above) and it would be better to change the warning below to an error, but...